telegram/telegram.py

```diff
@@ -0,0 +1,187 @@
+import json
+
+from treq.client import HTTPClient
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.web import http
+from twisted.web.client import Agent
+from twisted.web._newclient import ResponseFailed
+
+from vumi.transports.httprpc.httprpc import HttpRpcTransport
+from vumi.config import ConfigText, ConfigUrl
+
+
+class TelegramTransportConfig(HttpRpcTransport.CONFIG_CLASS):
+    bot_username = ConfigText(
+        'The username of our Telegram bot',
+        static=True,
+        required=True,
+    )
+    bot_token = ConfigText(
+        "Our bot's unique token to access the Telegram API",
+        static=True,
+        required=True,
+    )
+    outbound_url = ConfigUrl(
+        'The URL our bot should make requests to',
+        default='https://api.telegram.org/bot', static=True,
+    )
+    inbound_url = ConfigUrl(
+        'The URL our transport will listen on for Telegram updates',
+        static=True, required=True,
+    )
+
+
+class TelegramTransport(HttpRpcTransport):
+    """
+    Telegram transport for Vumi
+    """
+    transport_type = 'telegram'
+    transport_name = 'telegram_transport'
+
+    CONFIG_CLASS = TelegramTransportConfig
+
+    @classmethod
+    def agent_factory(cls):
+        """For swapping out the Agent for use in tests"""
+        return Agent(reactor)
+
+    @inlineCallbacks
+    def setup_webhook(self):
+        # NOTE: Telegram currently only supports ports 80, 88, 443 and 8443 for
+        #       webhook setup
+        url = self.get_outbound_url('setWebhook')
+        http_client = HTTPClient(self.agent_factory())
+
+        try:
+            r = yield http_client.post(
+                url=url,
+                data=json.dumps({'url': self.inbound_url}),
+                headers={'Content-Type': ['application/json']}
+                )
+        # Telegram redirects our request if our bot token is invalid
+        except ResponseFailed as e:
+            self.log.info(
+                'Webhook setup failed: Invalid token (redirected)\n%s' % e)
+            return
```

Comment:
If this is a problem that can cause our transport to not work, should it be logged at something a little louder than `info` level?